integration/smartbft/mock_orderer.go

```diff
@@ -184,19 +180,25 @@ func (mo *MockOrderer) deliverBlocks(
 			return status, nil
 		}
 
-		// increment block number to support FAIL_IF_NOT_READY deliver behavior
 		number++
-		mo.logger.Infof("[channel: %s] Delivering block [%d] for (%p) for %s", chdr.ChannelId, block.Header.Number, seekInfo, addr)
 
 		block2send := &cb.Block{
 			Header:   block.Header,
 			Metadata: block.Metadata,
 			Data:     block.Data,
 		}
+
 		if seekInfo.ContentType == ab.SeekInfo_HEADER_WITH_SIG && !protoutil.IsConfigBlock(block) {
-			mo.logger.Infof("ASKED FOR HEADER WITH SIG")
 			block2send.Data = nil
+		} else if mo.censorDataMode {
+			if !mo.peerFirstTry {
+				mo.malicious = true
+			}
+			if mo.sentCount >= 5 && mo.malicious {
+				break
```

Comment:
I don't think it is correct to break the loop here. You should not return from the DeliverBlocks loop but rather wait here until instructed to continue or exit the loop.
Exiting the loop here is like server failure, the BFTDeliverer will immediatlly choose another mock. What we need is for this server to remain silent but not break the connection.